### PR TITLE
Bluetooth: Mesh: Increase advertising thread stack size

### PR DESCRIPTION
**Pull request by erwango**
With BT_HOST_CRYPTO, advertising stack size could be overflowed,
increase size to 1024 when BT_HOST_CRYPTO is enabled.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

### CODE REVIEW

File: subsys/bluetooth/host/mesh/adv.c
Comment (jhedberg) [inline subsys/bluetooth/host/mesh/adv.c]: minor thing: we've tried to be consistent with `#if defined(CONFIG_...)` in the Bluetooth code, could you please do that? (benefit is that you can combine multiple `defined(...)` conditions for a single `#if` but not with `#ifdef`)

**Review by jhedberg — CHANGES_REQUESTED** [review]

**Review by jhedberg — APPROVED** [review]

**Comment by jhedberg** [comment]
@sjanc @Vudentz @carlescufi could we get a second review on this? A month ago or so it was discovered that the stack requirement for advertising had grown, see commit 1cfa12276c3723f0b84401454e65ba2e5956828b, however turns out we still need the conditionally higher size for BT_HOST_CRYPTO.

**Review by carlescufi — APPROVED** [review]
No objections

**Pull request merged**
